stdlib/email/src/main/ballerina/src/email/listener_endpoint.bal

```diff
@@ -0,0 +1,196 @@
+    # email:Error? result = emailListener.__stop();
+    # ```
+    #
+    # + return - () or else error upon failure to stop the listener
+    public function __stop() returns error? {
+        check self.stop();
+    }
+
+    # Binds a service to the `email:Listener`.
+    # ```ballerina
+    # email:Error? result = emailListener.__attach(helloService, hello);
+    # ```
+    #
+    # + s - Type descriptor of the service
+    # + name - Name of the service
+    # + return - `()` or else a `email:Error` upon failure to register the listener
+    public function __attach(service s, string? name) returns error? {
+        return self.register(s, name);
+    }
+
+    # Stops consuming messages and detaches the service from the `email:Listener`.
+    # ```ballerina
+    # email:Error? result = emailListener.__detach(helloService);
+    # ```
+    #
+    # + s - Type descriptor of the service
+    # + return - `()` or else a `email:Error` upon failure to detach the service
+    public function __detach(service s) returns error? {
+
+    }
+
+    # Stops the `email:Listener` forcefully.
+    # ```ballerina
+    # email:Error? result = emailListener.__immediateStop();
+    # ```
+    #
+    # + return - `()` or else a `email:Error` upon failure to stop the listener
+    public function __immediateStop() returns error? {
+        check self.stop();
+    }
+
+    # Stops the `email:Listener` gracefully.
+    # ```ballerina
+    # email:Error? result = emailListener.__gracefulStop();
+    # ```
+    #
+    # + return - () or else error upon failure to stop the listener
+    public function __gracefulStop() returns error? {
+        check self.stop();
+    }
+
+    function start() returns error? {
+        var scheduler = self.config.cronExpression;
+        if (scheduler is string) {
+            task:AppointmentConfiguration config = {appointmentDetails: scheduler};
+            self.appointment = new(config);
+        } else {
+            task:TimerConfiguration config = {intervalInMillis: self.config.pollingInterval, initialDelayInMillis: 100};
+            self.appointment = new (config);
+        }
+        var appointment = self.appointment;
+        if (appointment is task:Scheduler) {
+            check appointment.attach(appointmentService, self);
+            check appointment.start();
+        }
+        log:printInfo("Listening to remote server at " + self.config.host + "...");
+    }
+
+    function stop() returns error? {
+        var appointment = self.appointment;
+        if (appointment is task:Scheduler) {
+            check appointment.stop();
+        }
+        log:printInfo("Stopped listening to remote server at " + self.config.host);
+    }
+
+    # Polls the email server enspoint.
+    # ```ballerina
+    # email:Error? result = emailListener.poll();
+    # ```
+    #
+    # + return - () or else error upon failure to poll the listener
+    public function poll() returns error? {
+        return poll(self.config);
```

Comment:
Do we need to pass configuration and initialize Email client in every poll?

Reply:
Corrected the logic while restructuring the logic related to the configuration passing.